### PR TITLE
Add a pattern for making shared ownership explicit.

### PR DESCRIPTION
**Pull request by MaineC**
In cases where projects are already owned by multiple teams but without clear explicit Trusted Committers this pattern can help to bootstrap an initial set of TCs.

### CODE REVIEW

File: patterns/1-initial/explicit-shared-ownership.md
Comment (lenucksi) [inline patterns/1-initial/explicit-shared-ownership.md]: > An organisation is already using InnerSource best practices in several teams.

I did actually observe this mostly even without InnerSource being known or best practices applied. The effects described are in force nevertheless.

There is an extension to this here where a project is actively abandoned or it's existence is denied in fear of being hit with the maintenance while it still is a crucial part of a given system. The way to fix this involves the same route given here.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (lenucksi) [inline patterns/1-initial/explicit-shared-ownership.md:22]:
```suggestion
```

File: patterns/1-initial/explicit-shared-ownership.md
Comment (lenucksi) [inline patterns/1-initial/explicit-shared-ownership.md:33]:
```suggestion
```

File: patterns/1-initial/explicit-shared-ownership.md
Comment (lenucksi) [inline patterns/1-initial/explicit-shared-ownership.md:30]: Added this force as resulting from the review comment given above

```suggestion
There often is fear around maintenance efforts arising from unwanted attribution to the project

```

File: patterns/1-initial/explicit-shared-ownership.md
Comment (lenucksi) [inline patterns/1-initial/explicit-shared-ownership.md]: I agree, but think we should include the need to address the waves past the technical experts (i.e. line management etc.) resulting in making this more explicit to not block this sensible and outcome driven approach. In other words: Hint that there might be a need to handle the cheeses-of-all-sizes-game.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (lenucksi) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
```

**Review by lenucksi — APPROVED** [review]
Thanks for this excellent write-up @MaineC, you beat me to writing it up.
I can only confirm all of what is written up here as a really common pattern. 

I did add a few notes on additional observations.

As a hint: Reviewing would be a lot easier if individual sentences would start on individual lines given the GH review tooling.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
Explicit Shared Ownership
```

Using Titlecase for the title :)

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]: You also list "ownership is unclear" under Context.

As your pattern is helping to make the ownership clear, I think the above should only be listed in Forces but not in Context. Not 100% sure though.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
Teams are working independently but are providing one common platform as a service.
```

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]: If you were to shorten this to a 2 sentence description of the problem and solution, how would it sound like?

One try:
> A component that has seen organic adoption by multiple teams in an organization cannot be maintained by it's original authors anymore. Sharing the ownership for the component across the dependent teams in a transparent way removes ambiguity and helps the component to survive.

Maybe part of the text that gets stripped out in the process could be moved to the Problem section?

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
A volunteer creates an issue in the component's repository highlighting the apparent unclarity or even lack of ownership. Subsequently a volunteer (can be the same person) creates a suggestion for how the project should be developed going forward including a list of initial [Trusted Committers](../2-structured/trusted-committer.md). This suggestion is posted to the project's documentation (e.g. it's `README.md` file) as a pull request. This pull request is left open and shared with all people affected by the change. Feedback can be given and integrated asynchronously. Development of the final state is transparent for everyone.
```

Some formatting and a link to the TC pattern.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
The entire decision process backing the result is transparent and can be influenced by those affected, leading to higher buy-in for the final result. Also the argument leading to the final decisions are accessible for those new to the project.
```

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md:48]: With "setup" do you mostly mean the shared ownership process here?
I am wondering if we can be more explicit here.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md:17]: I would suggest to list all items in Context and Forces as bullet points, like a check list that people can go through.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md]: What is the cheeses-of-all-sizes-game? Is that an English idiom that I should know?

**Comment by MaineC** [comment]
> I did add a few notes on additional observations.

Feel free to commit those right away, anyone with commit access to this repo should have write access to the PR branch. (If not, let me know and I'll try to change that.)


> As a hint: Reviewing would be a lot easier if individual sentences would start on individual lines given the GH review tooling.

Thanks for the hint. This is coming just in time as I'm about to write another pattern. I'll (try to) remember that before submitting it for review here.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]: Before filing such an issue, should the volunteer do some type of safety check, to confirm if the repo is indeed unmaintained?

I am wondering what happens if somebody files an issue "Let's solve the lack of ownership on this repo", while the owners of the repo actually don't agree with that sentiment. Of course this all depends on the people involved. I am just exploring if something could be done here as a precaution so to say.

**Review by spier — CHANGES_REQUESTED** [review]
Great start @MaineC. I left some comments and suggestions.

Some general thoughts:
- how does this relate to how ownership is transitioned to a new maintainer in Open Source?
- how to determine whether shared ownership should be preferred over handing over responsibilities of the repo to a single team? (e.g. one of the teams that are depend on the component)

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md]: Trivial fix. Committing.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md:17]: This change seems non-controversial, so I will just make the change myself.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
- Teams are working independently but are providing one common platform as a service.

- The software they are working on has been evolving for a long time. Engineers working on sub-modules have changed over time.

- One component is in widespread use but ownership is unclear.
```

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md]: Trivial change. Committing.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
An organisation is already using InnerSource best practices in several teams. The architecture of the software offered has grown organically. 

While talking about code ownership and accountability, teams notice that there is a component that is in a worrying state: Developed by an original team of authors it has grown it's userbase way beyond what the original team can handle. From time to time others step up to help out, however there is no formal process established. As a result, conflicts arise around who should do the work, and who should decide on project direction. 

Simply forking the component would lead to a lot of duplication and wasted effort. Therefore the involved teams are looking for a less intrusive solution to the issue.
```

Splitting up the paragraph into sections, and making some wording and comma fixes.

**Comment by spier** [comment]
Hi @MaineC.

I merged and resolved a couple of feedback comments that I considered non-controversial.
Please pull the latest changes to see the most up-to-date version.

I was wondering if there is any other help that I can provide to get a first version of this merged and shared with a wider audience?

File: patterns/1-initial/explicit-shared-ownership.md
Comment (MaineC) [reply patterns/1-initial/explicit-shared-ownership.md]: I've tried to shorten the patlet without dropping too much information. Looking forward to your feedback.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (MaineC) [reply patterns/1-initial/explicit-shared-ownership.md]: I agree that this occurs also outside of the InnerSource context. I believe though that it is important to not start rolling out InnerSource by addressing this type of problem: The natural reflex in traditional corporations seems to be more a top down approach to solving this issue. The approach described in this pattern though is much more bottom up. I believe it needs a bit of experience to make this pattern work in practice.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (MaineC) [reply patterns/1-initial/explicit-shared-ownership.md]: I'm not against moving it only to the forces section. I lack the deep understanding of the pattern format to make the final call here.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (MaineC) [reply patterns/1-initial/explicit-shared-ownership.md:48]: I believe what I mean by setup is not only the who makes changes - but also how. Often in projects that are maintained locally in a single team there are many unwritten rules. When moving beyond the boundaries of a single team what I found helps a great deal is making these unwritten rules explicit. Things like

* who has to review
* how long should it be expected to take until a first pull request feedback
* who should approve, and how many Trusted Committers
* what coding guidelines should we follow
* what makes for valid concerns for rejecting pull requests or delaying them

... essentially all of the things that can lead to friction if left implicit (those can differ from team to team and from org to org).

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
An organisation is already using InnerSource best practices in several teams. The architecture of the software offered has grown organically.
```

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
While talking about code ownership and accountability, teams notice that there is a component that is in a worrying state: Developed by an original team of authors it has grown it's userbase way beyond what the original team can handle. From time to time others step up to help out, however there is no formal process established. As a result, conflicts arise around who should do the work, and who should decide on project direction.
```

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md]: Looks a lot better!

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md]: After reading this again, I think we can make it more clear by rewording the bullet under Context, as there the important piece is that the component is in widespread use, rather than that ownership is unclear (as the latter is the general problem that this pattern addresses.).

Will suggest a change in a separate thread.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
- Ownership of one component is unclear.

- There are too many teams dependent on the component to pull everyone into one single meeting to discuss ownership.

- There may be people dependent on the module that are not yet known.
```

Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md]: Formatting as bullet list. Likely non-controversial, hence I am merging this.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
There often is fear around maintenance efforts arising from unwanted attribution to the project.
```

Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]: About "There often is fear around maintenance efforts arising from unwanted attribution to the project":

Is this a force in its own right, i.e. a specific thing that makes solving the problem difficult?
In that case we should turn it into its own bullet, so that it becomes part of a checklist that readers can go through when reviewing the pattern.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
A volunteer creates an issue in the component's repository highlighting the apparent unclarity or even lack of ownership. Subsequently a volunteer (can be the same person) creates a suggestion for how the project should be developed going forward including a proposed list of initial [Trusted Committers](../2-structured/trusted-committer.md). This suggestion is posted to the project's documentation (e.g. it's `README.md` file) as a pull request. This pull request is left open and shared with all people affected by the change. Feedback can be given and integrated asynchronously. Development of the final state is transparent for everyone.
```

To make it more clear that the person opening the PR cannot just decide on the TCs but is rather proposing a solution that is then up for review. For people that know the PR review mechanisms that is likely clear, however it doesn't hurt to spell it out here.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
There is an initial team of [Trusted Committers](../2-structured/trusted-committer.md) committed to the component.
```

Linking here again doesn't hurt :)

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md]:
```suggestion
Expectations related to collaboration are transparent for everyone involved going forward e.g. by creating [base documentation in CONTRIBUTING.md](../2-structured/project-setup/base-documentation.md).
```

Is this link between the two patterns correct?

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md:48]: Sounds like things that might go to `CONTRIBUTING.md`?
I have another question about about that base documentation file, and if we should reference it from this pattern.

Feel free to resolve this thread.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [inline patterns/1-initial/explicit-shared-ownership.md:7]: What do you mean by "contributions document" here?
Are you referring to a CONTRIBUTING.md, or do you not want to be that prescriptive at this point?

**Review by spier — CHANGES_REQUESTED** [review]
Great next version again!

My only open question is how this pattern is connected to `CONTRIBUTING.md`.

If we can resolve that, then this pattern would be good be merged as Initial from my point of view.

As usual, thanks for spending the time to write this up!

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md]: Not essential. Resolving.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md]: @MaineC based on your last comment it would make sense to add this to the Context section, as this sounds like a requirement to successfully apply this pattern:

* The organisation is already using InnerSource best practices in several teams. (i.e. the ideas of InnerSource are not new for the organisation)

File: patterns/1-initial/explicit-shared-ownership.md
Comment (MaineC) [reply patterns/1-initial/explicit-shared-ownership.md]: https://innersourcecommons.org/documents/books/InnerSourceChecklist.pdf ... the big cheese story is explained on page 11 of this book. It comes up in several learning path trainings as well.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (MaineC) [reply patterns/1-initial/explicit-shared-ownership.md:7]: This should be referring to the CONTRIBUTING.md - though it really could be written down anywhere.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (MaineC) [reply patterns/1-initial/explicit-shared-ownership.md]: Hmm - wouldn't we have to add this to pretty much all of the patterns (except maybe the "start with an experiment" and similar ones)?

File: patterns/1-initial/explicit-shared-ownership.md
Comment (MaineC) [reply patterns/1-initial/explicit-shared-ownership.md]: To me this sentence is more of an explanation of why those dependant on this component are not more vocal about their dependency. But sure, we could also put that on it's own line.

**Comment by MaineC** [comment]
> how does this relate to how ownership is transitioned to a new maintainer in Open Source?

In my experience this is handled different going from project to project. What I have seen at Apache is a more structured approach that I believe would also make sense to adopt in an InnerSource setting. Instead of leaving downstream users in the dark as to whether projects are still active or not, here's what is done there:

* Projects have to send a minimal report to the board each quarter.
* In that report they are required to also talk about how many people there are who are still active in the PMC (still answering with "yes, I'm here, ready to take care of the project and actively watching it")
* If there are less than three of these people, the alarm bells start rining.
* The user list is informed that the project needs urgent help or will be moved to the attic instead (meaning that all communication channels will be moved to read only, no more releases, no new patches applied, no issues fixed)
* If no user steps up to help, the project is moved to the attic.

Before that happens, the project is nudged regularly to recruit new helping hands. Depending on the project those efforts may or may not be successful.

> how to determine whether shared ownership should be preferred over handing over responsibilities of the repo to a single team? 
> (e.g. one of the teams that are depend on the component)

At the end of the day this is specific to the situation at hand. In my experience this is something that those teams are well able to solve once the situation has been made transparent and stakeholders informed (in most cases they already know about the situation anyway).

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md]: Yeah, that is also true. So let's just leave things as they are here.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md]: I see. Let's leave it as is then.

File: patterns/1-initial/explicit-shared-ownership.md
Comment (spier) [reply patterns/1-initial/explicit-shared-ownership.md:48]: Thanks again for the extra info. I think this info could help to improve other patterns.
I created a new issue for that.

**Review by spier — APPROVED** [review]
LGTM

**Comment by spier** [comment]
As all inline comments are resolved now, I am opting to merge this pattern.

Thank you for the great contribution @MaineC. I am expecting that many more companies would know this situation, so I am curious to hear what other people will have to add to this in the future.

**Comment by spier** [comment]
When merging the PR I noticed the intro text of this PR:
> In cases where projects are already owned by multiple teams but without clear explicit Trusted Committers this pattern can help to bootstrap an initial set of TCs.

This is a great pitch for using the pattern actually. I am wondering if and how this could be worked into the pattern itself.

**Pull request merged**
